File: app/src/main/java/com/ymir/fern_app/ProfileActivity.java
package com.ymir.fern_app;

import java.net.MalformedURLException;
import java.util.List;
import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.os.AsyncTask;
import android.os.StrictMode;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.Toast;
import java.sql.*;
import java.util.concurrent.ExecutionException;

import com.microsoft.windowsazure.mobileservices.*;
import com.microsoft.windowsazure.mobileservices.table.MobileServiceJsonTable;
import com.microsoft.windowsazure.mobileservices.table.MobileServiceTable;

public class ProfileActivity extends AppCompatActivity {

    private Button mMapButton;
    private MobileServiceClient mClient;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_profile);

        mMapButton = findViewById(R.id.activity_profile_maps_button);
        mMapButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                switchToMaps();
            }
        });

        try {
            mClient = new MobileServiceClient(
                    "https://fernapp.azurewebsites.net",
                    this
            );

            MobileServiceTable<Person> mPersonTable = mClient.getTable(Person.class);
            List<Person> result = mPersonTable.execute().get();

        }
        catch (MalformedURLException | MobileServiceException | ExecutionException | InterruptedException e) {
            Log.w("TAGTAGTAG", "asdfadsadsa");
            e.printStackTrace();
        }
    }

    private void switchToMaps() {
        Intent intent = new Intent(this, MapsActivity.class);
        startActivity(intent);
    }


}

File: app/src/main/java/com/ymir/fern_app/ProfileActivity.kt
package com.ymir.fern_app

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.TextView
import com.ymir.fern_app.ActivitySwitcher.switchToEvents
import com.ymir.fern_app.ActivitySwitcher.switchToMap
import com.ymir.fern_app.ActivitySwitcher.switchToProfileCreate

class ProfileActivity : AppCompatActivity() {

    private lateinit var mMapButton: Button
    private lateinit var mEventButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_profile)

        var person:Person = Person.getInstance()
        if (person.firstName == null || person.bio == null) { switchToProfileCreate(applicationContext)}
        else { fillDetails(person) }

        mMapButton = findViewById(R.id.activity_profile_maps_button)
        mMapButton.setOnClickListener {view ->
            switchToMap(applicationContext, view)
        }

        mEventButton = findViewById(R.id.activity_profile_events_button)
        mEventButton.setOnClickListener {view ->
            switchToEvents(applicationContext, view)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_bar, menu)

        return true
    }

    fun fillDetails(person: Person) {
        var txt: TextView

        txt = findViewById(R.id.profile_activity_first_name)
        txt.setText(person.firstName)

        txt = findViewById(R.id.profile_activity_last_name)
        txt.setText(person.secondName)

        txt = findViewById(R.id.profile_activity_uni)
        txt.setText(person.university)

        txt = findViewById(R.id.profile_activity_degree)
        txt.setText(person.degree)

        txt = findViewById(R.id.profile_activity_campus)
        txt.setText(person.campus)

        txt = findViewById(R.id.profile_activity_bio)
        txt.setText(person.bio)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        var id = item?.itemId

        if (id == R.id.activity_profile_edit_text) {
            editProfile()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    fun editProfile() {
        switchToProfileCreate(applicationContext)
    }
}
